src/job.ts

```diff
@@ -395,7 +395,7 @@ export class Job {
         this._startTime = process.hrtime();
         const writeStreams = this.writeStreams;
         const reportsDotenvVariables = await this.initProducerReportsDotenvVariables(writeStreams, Utils.expandVariables(this._variables));
-        const expanded = Utils.expandVariables({...this._variables, ...reportsDotenvVariables});
+        const expanded = Utils.unscape$$Variables(Utils.expandVariables({...this._variables, ...reportsDotenvVariables}));
```

Comment:
seemed like we're `expandVariables` mutltiple times. 

Thus, moved out the unescape$$ logic to make sure that we don't unescape the `$$` too early.